packages/sdk/src/react/_internal/wagmi/__tests__/create-config.test.ts

```
@@ -0,0 +1,196 @@
+				},
+			};
+
+			const config = createWagmiConfig(marketplaceConfig, baseSdkConfig);
+			expect(config.connectors).toBeDefined();
+			expect(config.chains).toHaveLength(1);
+		});
+
+		it('should create SSR compatible config when ssr flag is true', () => {
+			const config = createWagmiConfig(
+				baseMarketplaceConfig,
+				baseSdkConfig,
+				true,
+			);
+			expect(config.storage).toBeDefined();
+			expect(config.chains).toHaveLength(1);
+		});
+	});
+
+	describe('failure cases', () => {
+		it('should throw error when trying to use embedded wallet without waasConfigKey', () => {
+			const marketplaceConfig: MarketplaceConfig = {
+				...baseMarketplaceConfig,
+				walletOptionsNew: {
+					connectors: ['walletconnect'],
+					includeEIP6963Wallets: false,
+					walletType: 'embedded',
+				},
+			};
+
+			const sdkConfig: SdkConfig = {
+				...baseSdkConfig,
+				wallet: {
+					embedded: {
+						waasConfigKey: '',
+						googleClientId: 'test-google-id',
+						appleClientId: 'test-apple-id',
+						appleRedirectURI: 'https://test.com/redirect',
+					},
+				},
+			};
+
+			expect(() =>
+				getWaasConnectors(marketplaceConfig, {
+					...sdkConfig,
+					wallet: {
+						...sdkConfig.wallet,
+						embedded: {
+							...sdkConfig.wallet?.embedded,
+							waasConfigKey: '', // Empty waasConfigKey should trigger the error
+						},
+					},
+				}),
+			).toThrow(MissingConfigError);
+		});
+
+		it('should still create config when walletConnectProjectId is missing', () => {
+			const marketplaceConfig: MarketplaceConfig = {
+				...baseMarketplaceConfig,
+				walletOptionsNew: {
+					connectors: ['walletconnect'],
+					includeEIP6963Wallets: true,
+					walletType: 'universal',
+				},
+			};
+
+			const config = createWagmiConfig(marketplaceConfig, baseSdkConfig);
+			expect(config.connectors).toBeDefined();
+			expect(config.chains).toHaveLength(1);
+		});
+
+		it('should still create config when projectAccessKey is empty', () => {
```

Current behaviour for `sdkConfig` without `projectAccessKey` is this. We need to check if it's an empty string and throw an error accordingly IMO, wdyt?

---

I think you are right we should just throw an error if the access key is undefined

---

I chose to handle it on Provider level, last 3 commits are about tests for Provider 👇🏼